### PR TITLE
Dont use fm stuff

### PR DESCRIPTION
**Pull request by proller**

### CODE REVIEW

File: src/script/lua_api/l_mainmenu.cpp [hunk 1]
```diff
@@ -1101,6 +1101,7 @@ int ModApiMainMenu::l_get_lan_servers(lua_State *L)
 	int top = lua_gettop(L);
 	unsigned int index = 1;
 
+	std::shared_lock lock(ServerList::lan_adv_client.mutex);
```
Comment (DustyBagel) [inline src/script/lua_api/l_mainmenu.cpp:1104]: Isn't mutex supposed to be unlocked at some point in this function?

**Comment by DustyBagel** [comment]
Not that it is that important because merging this would not effect this but you didn't remove the reference to the lock.cpp file which no longer exists in this pr, from the CMakeLists.txt file in the threading folder.

https://github.com/proller/minetest/blob/lock/src/threading/CMakeLists.txt#L5 should be removed.

File: src/script/lua_api/l_mainmenu.cpp (hunk 1 above)
Comment (proller) [reply src/script/lua_api/l_mainmenu.cpp:1104]: it will be unlocked in lock destructor

Comment (DustyBagel) [reply src/script/lua_api/l_mainmenu.cpp:1104]: Oh, ok, that makes sense.

**Pull request merged**